deps/lua/src/lua_cmsgpack.c

```diff
@@ -435,6 +435,7 @@ int table_is_an_array(lua_State *L) {
 
     stacktop = lua_gettop(L);
 
+    luaL_checkstack(L, 2, "in function table_is_an_array");
```

Comment:
don't we need to assert the return value?

Reply:
Unlike `lua_checkstack` that returns if it succeeded or not, this function will not return if failed and will just raise and error and use the Lua long jump machinism to handle the error.

Reply:
https://www.lua.org/manual/5.1/manual.html#luaL_checkstack